src/main/java/core/basesyntax/CircleSupplier.java

```diff
@@ -0,0 +1,13 @@
+package core.basesyntax;
+
+import java.util.Random;
+
+public class CircleSupplier {
+    private Random random = new Random();
+    private RandomSize size = new RandomSize();
+    private ColorSupplier color = new ColorSupplier();
+
+    public Figure getRandomCircle() {
+        return new Circle(size.getRandomSize(), color.getRandomColor());
+    }
+}

```

Comment:
The class name `CircleSupplier` suggests that its purpose is to supply Circle objects, but according to the checklist, we should avoid using verbs for class/interface names. Consider renaming it to something like `CircleFactory` or `CircleProvider` to better convey its intent.